### PR TITLE
🐛Fix clusterctl error when reading default config file

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR fixes https://github.com/kubernetes-sigs/cluster-api/issues/2982 and most specifically
1. makes clusterctl to configure rules for reading env variables no matter if the config file exists or not
2. fixes an error in checkDefaultConfig 

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/2982

### CODE REVIEW

**Comment by vincepri** [comment]
@fabriziopandini could you retitle the PR to be more explicit about what we're fixing? PR titles go in release notes

**Comment by fabriziopandini** [comment]
@JoelSpeed test are already verifying those code path, but injected values were hiding the problem  
But I'm not sure we can unit test this in isolation without injecting values

@wfernandes opinions?

**Comment by vincepri** [comment]
@wfernandes @fabriziopandini have we run a manual test to make sure the issue is now fixed? We should follow-up, or get better coverage for this edge case either with some unit tests or in one of the e2e tests

**Comment by fabriziopandini** [comment]
@vincepri I have, but hopefully @wfernandes can double check

**Comment by wfernandes** [comment]
I'll review this asap.

**Comment by fabriziopandini** [comment]
@wfernandes @vincepri I managed to get the test coverage for the error conditions this PR is addressing, so we should have a signal in case a regression is introduced

**Comment by wfernandes** [comment]
Reviewing

File: cmd/clusterctl/client/config/reader_viper_test.go [hunk 1]
```diff
@@ -156,6 +158,22 @@ func Test_viperReader_Get(t *testing.T) {
 	}
 }
 
+func Test_viperReader_GetWithoutDefaultConfig(t *testing.T) {
+	g := NewWithT(t)
+	dir, err := ioutil.TempDir("", "clusterctl")
+	g.Expect(err).NotTo(HaveOccurred())
+	defer os.RemoveAll(dir)
```
Comment (wfernandes) [inline cmd/clusterctl/client/config/reader_viper_test.go:165]: This directory is not needed here since the `viperReader` will have no `configPaths` for this test case.

**Comment by wfernandes** [comment]
@vincepri @fabriziopandini Do we need to backport this fix to the v0.3.4 that was just released?

**Comment by vincepri** [comment]
We'll need to release v0.3.5

File: cmd/clusterctl/client/config/reader_viper_test.go (hunk 1 above)
Comment (fabriziopandini) [reply cmd/clusterctl/client/config/reader_viper_test.go:165]: thanks for noticing this. now I'm using it so the test is isolated from the user config

**Comment by fabriziopandini** [comment]
squashed commits and addressed pending comment

**Comment by sb1975** [comment]
Is this working ? why I am getting error whether I specify in clusterctl.yaml or export as env variable..
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/805
